PARCEL_2_RFC.md

```diff
@@ -0,0 +1,1285 @@
+environment variable.
+
+#### `--https`
+
+This will generate a local certificate (which will be untrusted by your
+browser, you'll need to approve it) and serve your assets over `https://`
+
+##### `--cert <path>`
+
+Specify the filepath to your SSL certificate when using `--https`.
+
+##### `--key <path>`
+
+Specify the filepath to your SSL key when using `--https`.
+
+#### `--cache <dir>`, `--no-cache`
+
+Configure the cache directory with `--cache <dir>` or disable it altogether
+with `--no-cache`.
+
+#### `--hot`, `--no-hot`
+
+Turn hot reloading on or off.
+
+##### `--hot-hostname <hostname>`
+
+Configure the hot reloading hostname.
+
+##### `--hot-port <port>`
+
+Configure the hot reloading port.
+
+#### `--[no-]source-maps`
+
+Turn source maps on or off. Source maps are turned on by default except in
+production builds.
+
+#### `--autoinstall [npm/yarn], --no-autoinstall`
+
+When enabled, whenever Parcel discovers a dependency that isn't installed it
+will attempt to install it with either npm or Yarn (defaults to npm unless a
+`yarn.lock` exists).
+
+#### `--mode <mode>`
+
+Override the environment mode, including those manually configured to something
+else (ex. `--mode production` or `--mode development`). Defaults to the
+`NODE_ENV` environment variable.
+
+##### `--development, --dev`
+
+Aliases for `--mode development`.
+
+##### `--production, --prod`
+
+Aliases for `--mode production`.
+
+##### `--test`
+
+Aliases for `--mode test`.
+
+#### `--public-url <url>`
+
+[todo]
+
+#### `--log-level <level>`
+
+Set the log level, either "0" (no output), "1" (errors), "2" (warnings +
+errors) or "3" (all). (Default: 2)
+
+#### `--version, -v, -V`
+
+Return the current version of Parcel.
+
+#### `--help, -h`
+
+Get help with the CLI.
+
+## Parcel Config
+
+Parcel has always and will always work out of the box for many projects with
+zero configuration. It should always be extremely simple to get started. But if
+you do want more control, we give you the tools to do so.
+
+### Configuring external tools
+
+A huge part of what Parcel does is run other tools over your code. Instead of
+pulling all that configuration into Parcel, we make use of their own
+configuration systems. So if you're using Babel, you should just use `.babelrc`
+files to configure it.
+
+When we do need to introduce config, we create tool specific config files in
+order to do so.
+
+### Configuring Parcel
+
+When you do need to configure Parcel, it will be in one of 3 places.
+
+- If you need to configure the CLI, it will be a CLI flag
+- If you need to configure your package, it will be in the `package.json`
+- If you need to configure something with your files or the Parcel asset
+  pipeline, it will be in `.parcelrc`
+
+### `package.json`
+
+[todo]
+
+```json
+{
+  "name": "foo",
+  "main": "dist/main/index.js",
+  "module": "dist/module/index.js",
+  "browser": "dist/browser/index.js",
+  "browserslist": ["> 1%", "not dead"],
```

Comment:
I know it's already "common" to define "browserslist" as a top level property like this but it creates an inconsistency with the "engines" property. Does it make sense for "browsers" to be just another "engine"?

Reply:
If we change what people are already doing they will just end up adding and maintaining both